### PR TITLE
added filter deploy post and delete

### LINKED ISSUE #4182: Add filter POST deploy and filter DELETE deploy
#### Current Behavior

We do not have filter POST deploy and filter DELETE deploy 

we can take inspiration from [application deploy](https://github.com/meshery/meshery/blob/8fab9be65ee93510906bd39e166087d1717de64d/handlers/meshery_application_handler.go#L36) and [pattern deploy](https://github.com/meshery/meshery/blob/8fab9be65ee93510906bd39e166087d1717de64d/handlers/oam_handler.go#L49) (for some reason it is in oam_handler)  

docs ->  https://docs.meshery.io/reference/rest-apis#api-experimental-filter-get-handle-get-request-for-all-filters

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Shreyas220**
Signed-off-by: Shreyas220 <shreyas.ny@gmail.com>

**Description**
added ability filter deploy post and delete to the endpoint "/api/filter/deploy"

This PR fixes #4182

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: handlers/meshery_filter_handler.go [hunk 1]
```diff
@@ -252,3 +252,29 @@ func formatFilterOutput(rw http.ResponseWriter, content []byte, format string) {
 	rw.Header().Set("Content-Type", "application/json")
 	fmt.Fprint(rw, string(data))
 }
+
+// swagger:route POST /api/filter/deploy ApplicationsAPI idPostDeployFilterFile
```
Comment (utkarsh-pro) [inline handlers/meshery_filter_handler.go:256]:
```suggestion
// swagger:route POST /api/filter/deploy FilterAPI idPostDeployFilterFile
```

File: handlers/meshery_filter_handler.go [hunk 2]
```diff
@@ -252,3 +252,29 @@ func formatFilterOutput(rw http.ResponseWriter, content []byte, format string) {
...
+
+// FilterFileHandler handles the requested related to filter files
+func (h *Handler) FilterFileHandler(
+	rw http.ResponseWriter,
+	r *http.Request,
+	prefObj *models.Preference,
+	user *models.User,
+	provider models.Provider,
+) {
+	// Filter files are just pattern files?
```
Comment (utkarsh-pro) [inline handlers/meshery_filter_handler.go:278]: For now, yes :slightly_smiling_face:

**Review by utkarsh-pro — COMMENTED** [review]
Just a few nitpicks to address @Shreyas220

**Review by leecalcote — APPROVED** [review]

**Review by utkarsh-pro — DISMISSED** [review]

**Comment by Shreyas220** [comment]
@utkarsh-pro anything else ?

**Comment by leecalcote** [comment]
@navendu-pottekkat over to you for review.

**Review by leecalcote — APPROVED** [review]
LGTM.

@piyushsingariya, please add any feedback retroactively, if needed.

**Pull request merged**